pkg/manager/internal.go

```diff
@@ -179,6 +179,20 @@ func (cm *controllerManager) add(r Runnable) error {
 	return cm.runnables.Add(r)
 }
 
+// AddMetricsExtraHandler adds extra handler served on path to the http server that serves metrics.
+func (cm *controllerManager) AddMetricsExtraHandler(path string, handler http.Handler) error {
+	cm.Lock()
+	defer cm.Unlock()
+	if cm.started {
+		return fmt.Errorf("unable to add new metrics handler because metrics endpoint has already been created")
+	}
+	if err := cm.metricsServer.AddExtraHandler(path, handler); err != nil {
```

Comment:
I think metricsServer is nil if BindAddress = 0

I think we should avoid a panic here. Not sure if we want to just return nil or return an error. I think I'm leaning towards returning nil if cm.metricsServer is nil (because disabling the metrics by setting BindAddress to 0 could be an end-user choice, so the author of the controller might not have that under control)

Reply:
If the bind address is 0, the extra handlers should still be populated, but disabled, no?

Reply:
added check and log